files/scripts/supervisor-proc-exit-listener

```diff
@@ -62,52 +157,60 @@ def main(argv):
 
     critical_group_list, critical_process_list = get_critical_group_and_process_list()
 
+    process_under_alerting = {}
+    listener_ready = True
+
     while True:
         # Transition from ACKNOWLEDGED to READY
-        childutils.listener.ready()
-
-        line = sys.stdin.readline()
-        headers = childutils.get_headers(line)
-        payload = sys.stdin.read(int(headers['len']))
+        if listener_ready:
+            childutils.listener.ready()
+
+        stdin_instance = select.select([sys.stdin], [], [], SELECT_TIMEOUT_MINS)[0]
+        if len(stdin_instance) == 0:
+            for process in process_under_alerting.keys():
+                epoch_time = time.time()
+                if process_running(process):
+                    process_under_alerting.pop(process)
+                elif epoch_time - process_under_alerting[process] >= ALERTING_INTERVAL_SECS:
+                    process_under_alerting[process] = epoch_time
+                    generate_alerting_message(process)
+            # If there was no standard input from stdin, then continue to check and event listener 
+            # will still be in 'READY' state.
+            listener_ready = False
+            continue
+        else:
+            listener_ready = True
+        
+	#syslog.syslog(syslog.LOG_ERR, "Error message was NULL")
+        for instance in stdin_instance:
```

Comment:
Since you only passed one file descriptor, there can be at most one file descriptor in the returned list. There should be no need to loop here.